pgp/keysource.go

```diff
@@ -609,6 +588,27 @@ func gpgBinary() string {
 	return binary
 }
 
+// gnuPGHome determines the GnuPG home directory, and returns its path.
+// In order of preference:
+//  1. customPath
+//  2. $GNUPGHOME
+//  3. user.Current().HomeDir/.gnupg
+//  4. $HOME/.gnupg
+func gnuPGHome(customPath string) string {
+	if customPath == "" {
+		dir := os.Getenv("GNUPGHOME")
+		if dir == "" {
+			usr, err := user.Current()
+			if err != nil {
+				return filepath.Join(os.Getenv("HOME"), ".gnupg")
+			}
+			return filepath.Join(usr.HomeDir, ".gnupg")
+		}
+		return dir
+	}
+	return customPath
+}
```

Comment:
Nit: Any reason not to use early returns to match the order preference? IMO it improves readability 
```go
func gnuPGHome(customPath string) string {
  if customPath != "" {
    return customPath
  }

  dir := os.Getenv("GNUPGHOME")
  if dir != "" {
    return dir
  }

  usr, err := user.Current()
  if err != nil {
    return filepath.Join(os.Getenv("HOME"), ".gnupg")
  }

  return filepath.Join(usr.HomeDir, ".gnupg")
}
```

Reply:
No particular reason. I am in a weird camp that actually finds the current thing more readable as-is, but that is very subjective :-)